src/backend/access/transam/xloginsert.c

```diff
@@ -1181,6 +1181,7 @@ log_newpage_range(Relation rel, ForkNumber forkNum,
 			MarkBufferDirty(bufpack[i]);
 		}
 
+		//Zenith XXX: What's the purpose of this change?
```

Comment:
Zenith force WAL logging of evicted pages, so it can happen that in some cases when pages are first modified and then WAL logged (for example building GiST/GiN indexes) there are no more pages which need to be WAL logged at the end of build procedure. As far as XLogInsert throws error if not records were inserted, we need to add this check.